iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java

```diff
@@ -72,12 +74,19 @@ public void start() throws Exception {
      * Trigger prune on available disk is equal to or less than threshold
      */
     private final Runnable triggerPruneOnThresholdBreach = () -> {
-        while (true){
+        if (!isPruning) {
             long availableDiskPercentage = StatusReporter.getResourceConsumptionManagerStatus().getAvailableDisk() * 100 /
                     StatusReporter.getResourceConsumptionManagerStatus().getTotalDiskSpace();
-            if (Configuration.getAvailableDiskThreshold() >= availableDiskPercentage){
-                LoggingService.logDebug(MODULE_NAME, "Docker Prune when available disk is equal to or less than threshold");
-                docker.dockerPrune();
+            if (availableDiskPercentage < Configuration.getAvailableDiskThreshold()) {
+                try {
+                    LoggingService.logDebug(MODULE_NAME, "Docker Prune when available disk is less than threshold");
+                    isPruning = true;
+                    PruneResponse response = docker.dockerPrune();
+                    isPruning = false;
```

Comment:
set `isPruning` to `false` in `finally` block. Then there is no need to have it in both `try` and `catch`

Reply:
I always forget the finally block. Will update it thanks